libnetwork/drivers/bridge/internal/nftabler/network.go

```diff
@@ -4,29 +4,266 @@ package nftabler
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/containerd/log"
 	"github.com/docker/docker/libnetwork/drivers/bridge/internal/firewaller"
+	"github.com/docker/docker/libnetwork/internal/nftables"
+	"go.opentelemetry.io/otel"
 )
 
 type network struct {
-	config firewaller.NetworkConfig
-	fw     *nftabler
+	config      firewaller.NetworkConfig
+	cleanFuncs4 []func() error
+	cleanFuncs6 []func() error
+	fw          *nftabler
 }
 
 func (nft *nftabler) NewNetwork(ctx context.Context, nc firewaller.NetworkConfig) (_ firewaller.Network, retErr error) {
 	n := &network{
 		fw:     nft,
 		config: nc,
 	}
+	defer func() {
+		if retErr != nil {
+			if err := n.DelNetworkLevelRules(ctx); err != nil {
+				log.G(ctx).WithError(err).Error("Ignoring cleanup error")
+			}
+		}
+	}()
+
+	if n.fw.config.IPv4 {
+		var err error
+		n.cleanFuncs4, err = n.configure(ctx, nft.table4, n.config.Config4)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if n.fw.config.IPv6 {
+		var err error
+		n.cleanFuncs6, err = n.configure(ctx, nft.table6, n.config.Config6)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Ah, right, I had something like this in mind, because ultimately, the cleanup code (DelNetworkLevelRules) shouldn't have to be aware "what" it's cleaning up (ipv4, ipv6, "your room"), just that it has to be executed?

```go
c, err := n.configure(ctx, nft.table6, n.config.Config6)
if err != nil {
    return nil, err
}
n.cleanFuncs = append(n.cleanFuncs, c)
```


( `n.configure` could return a single `func() err` (do we need a context?), which in itself could be a chain of cleanup funcs)